internal/js/modules/k6/browser/tests/page_test.go

```diff
@@ -2182,3 +2182,273 @@ func TestPageOnMetric(t *testing.T) {
+				"accept-language": "en-US",
+				"referer":         tb.url("/home"),
+				"user-agent":      "some-user-agent",
+			},
+			FrameURL:             tb.url("/home"),
+			AcceptLanguageHeader: "en-US",
+			Headers: map[string]string{
+				"Accept-Language": "en-US",
+				"Referer":         tb.url("/home"),
+				"User-Agent":      "some-user-agent",
+			},
+			HeadersArray: []map[string]string{
+				{"name": "User-Agent", "value": "some-user-agent"},
+				{"name": "Accept-Language", "value": "en-US"},
+				{"name": "Referer", "value": tb.url("/home")},
+			},
+			IsNavigationRequest: false,
+			Method:              "GET",
+			PostData:            "",
+			PostDataBuffer:      "",
+			ResourceType:        "Stylesheet",
+			Size: map[string]int{
+				"body":    0,
+				"headers": 116,
+			},
+			URL: tb.url("/style.css"),
+		},
+		{
+			AllHeaders: map[string]string{
+				"accept-language": "en-US",
+				"content-type":    "application/json",
+				"referer":         tb.url("/home"),
+				"user-agent":      "some-user-agent",
+			},
+			FrameURL:             tb.url("/home"),
+			AcceptLanguageHeader: "en-US",
+			Headers: map[string]string{
+				"Accept-Language": "en-US",
+				"Content-Type":    "application/json",
+				"Referer":         tb.url("/home"),
+				"User-Agent":      "some-user-agent",
+			},
+			HeadersArray: []map[string]string{
+				{"name": "Referer", "value": tb.url("/home")},
+				{"name": "User-Agent", "value": "some-user-agent"},
+				{"name": "Accept-Language", "value": "en-US"},
+				{"name": "Content-Type", "value": "application/json"},
+			},
+			IsNavigationRequest: false,
+			Method:              "POST",
+			PostData:            `{"name":"tester"}`,
+			PostDataBuffer:      `{"name":"tester"}`,
+			ResourceType:        "Fetch",
+			Size: map[string]int{
+				"body":    17,
+				"headers": 143,
+			},
+			URL: tb.url("/api"),
+		},
+		{
+			AllHeaders: map[string]string{
+				"accept-language": "en-US",
+				"referer":         tb.url("/home"),
+				"user-agent":      "some-user-agent",
+			},
+			FrameURL:             tb.url("/home"),
+			AcceptLanguageHeader: "en-US",
+			Headers: map[string]string{
+				"Accept-Language": "en-US",
+				"Referer":         tb.url("/home"),
+				"User-Agent":      "some-user-agent",
+			},
+			HeadersArray: []map[string]string{
+				{"name": "Accept-Language", "value": "en-US"},
+				{"name": "Referer", "value": tb.url("/home")},
+				{"name": "User-Agent", "value": "some-user-agent"},
+			},
+			IsNavigationRequest: false,
+			Method:              "GET",
+			PostData:            "",
+			PostDataBuffer:      "",
+			ResourceType:        "Other",
+			Size: map[string]int{
+				"body":    0,
+				"headers": 118,
+			},
+			URL: tb.url("/favicon.ico"),
+		},
+	}
+
+	// Compare each request one by one for better test failure visibility
+	for _, req := range requests {
+		i := -1
+		for j, e := range expected {
+			if req.URL == e.URL {
+				i = j
+				break
+			}
+		}
```

Comment:
We can shorten this to:

```suggestion
		i := slices.IndexFunc(expected, func(r request) bool { return req.URL == r.URL })
		assert.NotEqual(t, -1, i, "failed to find expected request with URL %s", req.URL)
```

Reply:
TIL! Thanks 😄  refactored in 8d5b9d11217c403e3f2f4d5ef8ffbc078e51af21